### PR TITLE
[fs tests] refactor and extend test cases

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR:
- Refactors and extends the test cases for the HTTP File System related functionality.
- Fixes the docstrings for the AZFS test cases.
- Switches to compatible tf public api's for AZFS and HTTP FS tests.

### CODE REVIEW

File: tests/test_http_eager.py [hunk 1]
```diff
@@ -15,38 +15,72 @@
...
-        i += 1
-    assert i == len(lines)
-
-    remote = tf.io.gfile.GFile(filename)
-    assert remote.size() == len(local)
-    offset_start = list(range(0, len(local), 100))
-    offset_stop = offset_start[1:] + [len(local)]
-    for (start, stop) in zip(offset_start, offset_stop):
-        assert remote.read(100) == local[start:stop]
+class HTTPFSTest(test.TestCase):
```
Comment (yongtang) [inline tests/test_http_eager.py:23]: It looks like `tf.test.TestCase` is public API. I think we can use that directly like below?
```
class HTTPFSTest(tf.test.TestCase):
```


(The `from tensorflow.python.platform import test` is not publicly exposed endpoint so they might be subject to change in the future.

File: tests/test_azure.py [hunk 2]
```diff
@@ -26,9 +26,9 @@
 class AZFSTest(test.TestCase):
```
Comment (yongtang) [inline tests/test_azure.py:26]: That was from the past code but I think we can use `tf.test.TestCase` as well.

Also, `tf.io.gfile` can be used directly. So at the beginning of the file:
```
from tensorflow.python.platform import test
from tensorflow.python.platform import gfile
```

can be replaced with:
```
import tensorflow as tf
```

File: tests/test_http_eager.py (hunk 1 above)
Comment (kvignesh1420) [reply tests/test_http_eager.py]: addressed in the commit

File: tests/test_azure.py (hunk 2 above)
Comment (kvignesh1420) [reply tests/test_azure.py]: addressed in the commit

**Comment by kvignesh1420** [comment]
@yongtang I have observed that the public API for tf.io.gfile has changed a bit.

For example:

in current test_azure.py
```python
def setUp(self):  # pylint: disable=invalid-name
        super().setUp()
        if not tf.io.gfile.IsDirectory(self.path_root):
            tf.io.gfile.MakeDirs(self.path_root)
```

But, the prompts for the API's are a bit different:

image

**Comment by yongtang** [comment]
@kvignesh1420 The APIs that are listed on tensorflow.org's api_docs (e.g., https://www.tensorflow.org/api_docs/python/tf/io/gfile/isdir) are considered as `public API` and have the guarantee of API-compatibility.

Alternatively, the symbols inside tensorflow's code base with `@tf_export("...")` decorated are considered as `public API` (tensorflow.org only extracts APIs defined with `@tf_export("...")`):
https://github.com/tensorflow/tensorflow/blob/fcc4b966f1265f466e82617020af93670141b009/tensorflow/python/lib/io/file_io.py#L499

Note in the code base you may notice some API specified with `v1` (e.g., `@tf_export(v1=["gfile.Copy"])`):
https://github.com/tensorflow/tensorflow/blob/v2.3.1/tensorflow/python/lib/io/file_io.py#L483

Those APIs are only for TF 1.x and will not be exposed with TF 2.0.

**Comment by yongtang** [comment]
Relevant to this PR we should use the ones that are exposed, i.e., instead of `tf.io.gfile.MakeDirs`, we should use `tf.io.gfile.makedirs` (public APIs), as `tf.io.gfile.MakeDirs` does not guarantee no future changes.

**Comment by kvignesh1420** [comment]
@yongtang thanks for the info. Made the applicable changes. PTAL.

**Review by yongtang — APPROVED** [review]

**Pull request merged**